internal/db/storage/blob/s3/s3_test.go

```diff
@@ -50,42 +60,136 @@ func TestNew(t *testing.T) {
 		if !errors.Is(err, w.err) {
 			return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
 		}
-		if !reflect.DeepEqual(got, w.want) {
-			return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+
+		opts := []comparator.Option{
+			comparator.AllowUnexported(client{}),
+			comparator.Comparer(func(want, got errgroup.Group) bool {
+				return reflect.DeepEqual(want, got)
+			}),
+			comparator.Comparer(func(want, got *s3.S3) bool {
+				return want != nil && got != nil
```

Comment:
should we comparator this object more accurately ?

Reply:
For example, what kind of information do you want to comparator?

Reply:
I think it is better to compare the internal S3 status and information to ensure the internal S3 status is the same as we expect.

Reply:
thank you.
but we can not use reflect.DeepEqual. because s3 contains function type field.
so I think it's enough to compare some of the information. What do you think?

Reply:
I am not sure but I wrote all the comparator for the struct. 
Do you think is it over-compare the struct?

Reply:
thank you. I resolve it.
waht do you think the following implementation??
https://github.com/vdaas/vald/pull/837/commits/250df9c76d00a4648dc0223cd74744061ff20fb1